src/resources/views/crud/fields/radio.blade.php

```diff
@@ -1,6 +1,7 @@
 <!-- radio -->
 @php
-    $optionValue = old(square_brackets_to_dots($field['name'])) ?? $field['value'] ?? $field['default'] ?? '';
+    $optionValue = old(square_brackets_to_dots($field['name'])) ?? $field['value'] ?? $field['default'] ?? 0;
```

Comment:
Having `0` as a fallback value is not a good idea in my opinion. It's a very common value for options and it's not ok to assume that's what the developer wants as a `default`. If the developer hasn't specified a `default`, it's best to assume there should be no `default`.

Reply:
```
$options = $field['options'];
$defaultValue = $field['default'] ?? (count($options) > 0 ? array_key_first($options) : '');
$fieldValue = !isset($field['value']) ? $defaultValue : (is_bool($field['value']) ? (int)$field['value'] : $field['value']);
$oldValue = old(square_brackets_to_dots($field['name']));
$currentValue = is_null($oldValue) ? $fieldValue : (is_bool($oldValue) ? (int)$oldValue : $oldValue);
$optionValue = $currentValue;
```
I realized my mistake, and I think it will be more logical and universal, if we take in the absence of any value, take the first key from **options**, and if the existing value is a _boolean_, then we will bring it to a number (I don't know, but **false** doesn't perceived).